src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
 
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = LocalDate.parse(dateFrom, DATE_FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, DATE_FORMATTER);
+        StringBuilder result = new StringBuilder();
+
+        result.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+
+        int[] salaries = new int[names.length];
+
+        for (String entry : data) {
+            String[] parts = entry.split(" ");
+            LocalDate entryDate = LocalDate.parse(parts[0], DATE_FORMATTER);
+            String employeeName = parts[1];
+            int hoursWorked = Integer.parseInt(parts[2]);
+            int payPerHour = Integer.parseInt(parts[3]);
+
+            if (!entryDate.isBefore(startDate) && !entryDate.isAfter(endDate)) {
+                for (int i = 0; i < names.length; i++) {
+                    if (names[i].equals(employeeName)) {
+                        salaries[i] += hoursWorked * payPerHour;
+                    }
+                }
+            }
+        }
+
+        for (int i = 0; i < names.length; i++) {
+            result.append(names[i])
+                    .append(" - ")
+                    .append(salaries[i])
+                    .append(System.lineSeparator());
+        }
+
+        return result.toString().trim();
```

Comment:
Instead of using `result.toString().trim()`, you could avoid appending the last line separator in the first place. This would make your code a bit cleaner and more efficient.